src/test-helpers.tsx

```diff
@@ -928,7 +928,7 @@ export const dispatchComment = ({
   comment = createFakeExternalComment(),
   store = configureStore(),
   ...params
-} = {}) => {
```

Comment:
Since it's possible to still call `dispatchComment()` without arguments, can we keep this code as it was? Were you trying to make the code coverage report happy? If so, I wouldn't worry about it. I think there is a bug in codecov where it doesn't recognize the ignore flag in comments. Also, high code coverage isn't so important because we use TypeScript 💪

Reply:
I was trying to make TSC happy. TSC complains when calling `dispatchComment({comment,  store, versionId})`. Error message is something like: 

> Object literal may only specify known properties, and 'versionId' does not exist in type '{ comment?: { filename: string | null; id: number; lineno: number | null; canned_response: ExternalCannedResponse | null; comment: string | null; user: { ...; } | { ...; }; version: ExternalVersionWithContent; } | undefined; store?: Store<...> | undefined; }'.
> 
> 539           versionId,

It looks like that `...params` cannot work as expected, since TS does not allow other object properties than `comment` and `store`. Maybe we can just remove `...params`?
```diff
diff --git a/src/test-helpers.tsx b/src/test-helpers.tsx
index ae6d0b1..7900ec7 100644
--- a/src/test-helpers.tsx
+++ b/src/test-helpers.tsx
@@ -927,9 +927,9 @@ export const dispatchComment = ({
   /* istanbul ignore next */
   comment = createFakeExternalComment(),
   store = configureStore(),
-  ...params
-}) => {
-  return dispatchComments({ comments: [comment], store, ...params });
+  versionId = nextUniqueId(),
+} = {}) => {
+  return dispatchComments({ comments: [comment], store, versionId });
 };
 
 export const createStoreWithVersionComments = ({
```

Reply:
Oh, right, I see. This was because those two functions were defined lazily, relying on default values to define the types. It is now time to set it up explicitly:

```diff
diff --git a/src/test-helpers.tsx b/src/test-helpers.tsx
index ae6d0b1..ba07133 100644
--- a/src/test-helpers.tsx
+++ b/src/test-helpers.tsx
@@ -912,12 +912,18 @@ export const createFakeComment = (comment: Partial<Comment> = {}) => {
 
 export const fakeAction = createAction('FAKE_ACTION');
 
+type DispatchCommentsParams = {
+  store?: Store;
+  comments?: ExternalComment[];
+  versionId?: number;
+};
+
 export const dispatchComments = ({
   store = configureStore(),
   /* istanbul ignore next */
   comments = [createFakeExternalComment()],
   versionId = 1,
-} = {}) => {
+}: DispatchCommentsParams = {}) => {
   store.dispatch(commentsActions.setComments({ versionId, comments }));
 
   return { store };
@@ -928,7 +934,9 @@ export const dispatchComment = ({
   comment = createFakeExternalComment(),
   store = configureStore(),
   ...params
-}) => {
+}: { comment?: ExternalComment } & Partial<
+  Omit<DispatchCommentsParams, 'comments'>
+> = {}) => {
   return dispatchComments({ comments: [comment], store, ...params });
 };
 
```

I am suggesting `Partial<...>` because that's more true to what an object spread is. It would only be necessary if `DispatchCommentsParams` grew required props in the future.

There is more info on utility types here: https://www.typescriptlang.org/docs/handbook/utility-types.html